### PR TITLE
Local Adapter with customizable Folder list

### PR DESCRIPTION
**Pull request by kasvith**
### Summary of Changes
This is a fellow up PR for #282 
Please resolve it before proceed.

- Updated Local Adapter to use a folder list
image

- Updated Translations for new changes

### Testing Instructions
- Go to Filesystem - Local plugin
- Under directories add any directory  you want
- Save
- Go to Media Manager

### Expected result
- Added directories should be mounted

### CODE REVIEW

File: plugins/filesystem/local/local.xml [hunk 1]
```diff
@@ -29,6 +31,19 @@
...
+					name="directories"
+					type="subform"
+					label="PLG_FILESYSTEM_LOCAL_DIRECTORIES_LABEL"
+					description="PLG_FILESYSTEM_LOCAL_DIRECTORIES_DESC"
+					multiple="true"
+					layout="joomla.form.field.subform.repeatable"
+					buttons="add,remove,move"
+				>
+					<form>
+						<field name="directory" type="folderlist" default="images" label="Directory" filter="" exclude="" stripext="" />
```
Comment (laoneo) [inline plugins/filesystem/local/local.xml]: String needs to be translated.

File: plugins/filesystem/local/local.xml [hunk 2]
```diff
@@ -29,6 +31,19 @@
 					description="PLG_FILESYSTEM_LOCAL_DISPLAY_NAME_DESC"
 					default="Local"
 				/>
+				<field
+					name="directories"
+					type="subform"
+					label="PLG_FILESYSTEM_LOCAL_DIRECTORIES_LABEL"
+					description="PLG_FILESYSTEM_LOCAL_DIRECTORIES_DESC"
+					multiple="true"
+					layout="joomla.form.field.subform.repeatable"
```
Comment (laoneo) [inline plugins/filesystem/local/local.xml]: Probably the table layout is nicer

File: plugins/filesystem/local/local.xml [hunk 3]
```diff
@@ -12,6 +12,8 @@
 	<files>
 		<filename plugin="local">local.php</filename>
 		<folder>Adapter</folder>
+		<folder>forms</folder>
+		<filename>forms/directory_list_form.xml</filename>
```
Comment (laoneo) [inline plugins/filesystem/local/local.xml]: Is this needed?

File: plugins/filesystem/local/local.php [hunk 4]
```diff
@@ -37,11 +37,26 @@ class PlgFileSystemLocal extends CMSPlugin
...
 	public function onFileSystemGetAdapters()
 	{
-		// Compile the root path
+		$adapters = [];
+
+		// Compile the root path and load default images folder
 		$filePath = ComponentHelper::getParams('com_media')->get('file_path', 'images');
 		$root = JPATH_ROOT . '/' . $filePath;
 		$root = rtrim($root) . '/';
+		$adapters[] = new \Joomla\Plugin\Filesystem\Local\Adapter\LocalAdapter($root, $filePath);
```
Comment (laoneo) [inline plugins/filesystem/local/local.php]: Add images as default value instead of hardcoding it, perhaps somebody doesn't want to have images.

**Review by laoneo — CHANGES_REQUESTED** [review]

File: plugins/filesystem/local/local.xml (hunk 3 above)
Comment (kasvith) [reply plugins/filesystem/local/local.xml]: old artifacts lol

File: plugins/filesystem/local/local.xml (hunk 2 above)
Comment (kasvith) [reply plugins/filesystem/local/local.xml]: done

File: plugins/filesystem/local/local.xml (hunk 1 above)
Comment (kasvith) [reply plugins/filesystem/local/local.xml]: done

File: plugins/filesystem/local/local.php (hunk 4 above)
Comment (kasvith) [reply plugins/filesystem/local/local.php]: so shall we remove option from com_media ? we get images from there

Comment (laoneo) [reply plugins/filesystem/local/local.php]: yes

File: plugins/filesystem/local/local.php [hunk 5]
```diff
@@ -37,11 +37,27 @@ class PlgFileSystemLocal extends CMSPlugin
 	 */
 	public function onFileSystemGetAdapters()
 	{
-		// Compile the root path
-		$filePath = ComponentHelper::getParams('com_media')->get('file_path', 'images');
-		$root = JPATH_ROOT . '/' . $filePath;
-		$root = rtrim($root) . '/';
+		$adapters = [];
+		$directories = $this->params->get('directories', []);
```
Comment (laoneo) [inline plugins/filesystem/local/local.php]: The default value must be here also images, when not set to be consistent with the UI.

Comment (kasvith) [reply plugins/filesystem/local/local.php]: The return value from subform is a json string and we parse it below. I'll fix it

**Comment by laoneo** [comment]
Thanks.

**Pull request merged**